core/modules/shrscan/Director.h

```diff
@@ -0,0 +1,292 @@
+        bool               isCanceld; //< When true is cancelling
+    };
+
+    virtual Status getStatus(Query const* query) = 0;
+    //-----------------------------------------------------------------------------
+    //! @brief Resume a possibly suspended shared scan query.
+    //!
+    //! When a query is resumed it becomes eligible to run when sufficient
+    //! resources become available.
+    //!
+    //! @param  query - Pointer to a Query object that describes the query.
+    //!
+    //! @return false: The query was not found.
+    //! @return true:  The query was resumed.
+    //-----------------------------------------------------------------------------
+
+    virtual bool  resumeQuery(Query const* query) = 0;
+
+    //-----------------------------------------------------------------------------
+    //! @brief Resume all suspended shared scan queries.
+    //!
+    //! This method effectively calls resumeQuery() on each suspended query.
+    //-----------------------------------------------------------------------------
+
+    virtual void  resumeAll() = 0;
+
+    //-----------------------------------------------------------------------------
+    //! @brief Suspend a shared scan query.
+    //!
+    //! When a running query is suspended, it is placed in the eligible to run
+    //! queue in FIFO order. If the query is executing its runQuery() method,
+    //! the suspension occurs upon return.
+    //!
+    //! @param  query   - Pointer to the Query object that describes the query.
+    //! @param  unlock  - When true, any additional tables required by the query
+    //!                   that were locked are unlocked. Otherwise, the tables
+    //!                   remain locked which may prevent new queries to run.
+    //!
+    //! @return false: The query was not found.
+    //! @return true:  The query was suspended or scheduled to be suspended.
+    //-----------------------------------------------------------------------------
+
+    virtual bool  suspendQuery(Query const* query, bool unlock=true) = 0;
+
+    //-----------------------------------------------------------------------------
+    //! @brief Suspend all shared scan queries.
+    //!
+    //! This method effectively calls suspendQuery() on each query (see above).
+    //!
+    //! @param  unlock  - When true, any additional tables required by the query
+    //!                   that were locked are unlocked. Otherwise, the tables
+    //!                   remain locked which may prevent new queries to run.
+    //-----------------------------------------------------------------------------
+
+    virtual void  suspendAll(bool unlock=true) = 0;
+
+    //-----------------------------------------------------------------------------
+
+    Director & operator=(const Director&) = delete;
+    Director(const Director&) = delete;
+
+                  Director() {}
+    virtual      ~Director() {}
+};
+
+}}} // namespace lsst:qserv:shrscan
+#endif
```

Comment:
We usually put a comment with the name of the include guard here as well, to make it clear that this #endif belongs to the include guard, e.g.:

```C++
#endif // LSST_QSERV_SHRSCAN_DIRECTOR_H
```

Also, prefer a trailing newline since some compilers bitch/warn about that, believe it or not...